src/content/blog/why-not-effect.mdx

```diff
@@ -0,0 +1,43 @@
+---
+title: "Why Val Town is not using Effect"
+description: The very cool functional programming library that could improve our codebase but is too much for our onboarding
+pubDate: October 23, 2024
+author: Tom MacWright
+---
+
+Val Town's backend is a pretty complex application. We're running user code, proxying blob storage and other services, storing observability data, running intervals, and a lot more.
+
+One of the things that makes it complicated is the wide variety of error states. There are simple errors, like when a client supplies us with an invalid input, but there are a lot of more sophisticated errors, like when a Val has invalid dependencies or syntax, and then when Vals fail with errors when operating as HTTP endpoints or cron triggers. Distinguishing between internal and external, recoverable and non-recoverable errors, is pretty crucial.
+
+We also have some pretty sophisticated asynchronous code. Vals should run as fast as possible, which means that we parallelize some operations and defer others. There are some pretty tricky functions that string it all together.
```

Comment:
Could be nice if you could make this more concrete in one of two ways:

1. Show some (simplified version of) the tricky nested code we have there
2. Explain some of the tricky errors we've been facing because of this code

Could also be interesting to mention the context of bringing new folks onto the team, which you allude to throughout

Reply:
Similarly, could be interesting to folks to actually show that same code then Effect-ified so they can get a feel for the difference and possible improvement, but also how foreign it is